src/Cli/dotnet/Installer/Windows/MsiPackageCache.cs

```diff
@@ -88,41 +88,63 @@ internal class MsiPackageCache : InstallerBase
         }
 
         /// <summary>
-        /// Creates the root package cache directory if it does not exist and configures the directory's ACLs. ACLs are still configured
-        /// if the directory exists.
-        /// </summary>
-        private void CreateRootDirectory()
-        {
-            CreateSecureDirectory(PackageCacheRoot);
-        }
-
-        /// <summary>
-        /// Creates the specified directory and secures it by configuring access rules (ACLs). If the parent
-        /// of the directory does not exist, it recursively walks the path back to ensure each parent directory
-        /// is created with the proper ACLs and inheritance settings.
+        /// Creates the specified directory and secures it by configuring access rules (ACLs) that allow sub-directories
+        /// and files to inherit access control entries. 
         /// </summary>
         /// <param name="path">The path of the directory to create.</param>
-        private void CreateSecureDirectory(string path)
+        /// <param name="resetAccessRules">When <see langword="true"/>, all existing access rules are removed before
+        /// setting up new rules.
+        /// </param>
+        /// <param name="log">The underlying setup log to use.</param>
+        public static void CreateSecureDirectory(string path, bool resetAccessRules = false, ISetupLogger log = null)
         {
             if (!Directory.Exists(path))
             {
-                CreateSecureDirectory(Directory.GetParent(path).FullName);
+                // Create the directory with the desired access rights if it doesn't exist. We don't use
+                // Directory.CreateDirectory as that can cause problems with inherited ACEs from
```

Comment:
I'm a bit confused here. If we inherit the parent folder's access rules, then reset them, that should be the equivalent of creating it anew, right?

Regardless, it would be nice to unify as much fo the DirectorySecurity modifications as possible 🙂

Reply:
I've removed the cleanup code, so things are much simpler now. Even if we modified the top-level permissions, it won't translate through to all the child objects.